Tests/PriorityQueueTests/HeapTests.swift

```diff
@@ -0,0 +1,426 @@
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+@testable import PriorityQueueModule
+
+final class HeapTests: XCTestCase {
+  func test_isEmpty() {
+    var heap = Heap<Int>()
+    XCTAssertTrue(heap.isEmpty)
+
+    heap.insert(42)
+    XCTAssertFalse(heap.isEmpty)
+
+    let _ = heap.popMin()
+    XCTAssertTrue(heap.isEmpty)
+  }
+
+  func test_count() {
+    var heap = Heap<Int>()
+    XCTAssertEqual(heap.count, 0)
+
+    heap.insert(20)
+    XCTAssertEqual(heap.count, 1)
+
+    heap.insert(40)
+    XCTAssertEqual(heap.count, 2)
+
+    _ = heap.popMin()
+    XCTAssertEqual(heap.count, 1)
+  }
+
+  func test_unordered() {
+    let heap = Heap<Int>((1...10))
+    XCTAssertEqual(Set(heap.unordered), Set(1...10))
+  }
+
+  struct Task: Comparable {
+    let name: String
+    let priority: Int
+
+    static func < (lhs: Task, rhs: Task) -> Bool {
+      lhs.priority < rhs.priority
+    }
+  }
+
+  func test_insert() {
+    var heap = Heap<Task>()
+
+    XCTAssertEqual(heap.count, 0)
+    heap.insert(Task(name: "Hello, world", priority: 50))
+    XCTAssertEqual(heap.count, 1)
+  }
+
+  func test_insert_contentsOf() {
+    var heap = Heap<Int>()
+    heap.insert(contentsOf: (1...10).shuffled())
+    XCTAssertEqual(heap.count, 10)
+    XCTAssertEqual(heap.popMax(), 10)
+    XCTAssertEqual(heap.popMin(), 1)
+
+    heap.insert(contentsOf: (21...50).shuffled())
+    XCTAssertEqual(heap.count, 38)
+    XCTAssertEqual(heap.max(), 50)
+    XCTAssertEqual(heap.min(), 2)
+
+    heap.insert(contentsOf: [-10, -9, -8, -7, -6, -5].shuffled())
+    XCTAssertEqual(heap.count, 44)
+    XCTAssertEqual(heap.min(), -10)
+  }
+
+  func test_insert_contentsOf_withSequenceFunction() {
+    func addTwo(_ i: Int) -> Int {
+      i + 2
+    }
+
+    let evens = sequence(first: 0, next: addTwo(_:)).prefix(20)
+    var heap = Heap(evens)
+    XCTAssertEqual(heap.count, 20)
+
+    heap.insert(contentsOf: sequence(first: 1, next: addTwo(_:)).prefix(20))
+    XCTAssertEqual(heap.count, 40)
+
+    for (idx, i) in heap.ascending.enumerated() {
+      XCTAssertEqual(idx, i)
+    }
+  }
+
+  func test_min() {
+    var heap = Heap<Int>()
+    XCTAssertNil(heap.min())
+
+    heap.insert(5)
+    XCTAssertEqual(5, heap.min())
+
+    heap.insert(12)
+    XCTAssertEqual(5, heap.min())
+
+    heap.insert(2)
+    XCTAssertEqual(2, heap.min())
+
+    heap.insert(1)
+    XCTAssertEqual(1, heap.min())
+  }
+
+  func test_max() {
+    var heap = Heap<Int>()
+    XCTAssertNil(heap.max())
+
+    heap.insert(42)
+    XCTAssertEqual(42, heap.max())
+
+    heap.insert(20)
+    XCTAssertEqual(42, heap.max())
+
+    heap.insert(63)
+    XCTAssertEqual(63, heap.max())
+
+    heap.insert(90)
+    XCTAssertEqual(90, heap.max())
+  }
+
+  func test_popMin() {
```

Comment:
FIXME: Combinatorial tests would work great on these!

This doesn't need to be addressed before this lands.

Reply:
Issue filed as https://github.com/apple/swift-collections/issues/73